pkg/apis/agones/v1/gameserver.go

```diff
@@ -389,8 +389,12 @@ func (gss GameServerSpec) Validate(devAddress string) ([]metav1.StatusCause, boo
 			})
 		}
 	}
-	return causes, len(causes) == 0
+	objMetaCauses := validateObjectMeta(gss.Template.ObjectMeta)
```

Comment:
Should the changes to this file (and gameserverset.go) have unit tests akin to what you added in fleet_test.go?

Reply:
I have added similar test for gameserver .